tests/classes/Variables.js

```diff
@@ -53,6 +53,25 @@ describe('Service', () => {
       serverless.variables.populateVariable.restore();
     });
 
+    it('should call overwrite if overwrite syntax provided and first option is self', () => {
```

Comment:
This test is actually not testing the behaviour, e.g. if you revert variables.js this test will pass. This behaviour only happens for nested `self:`, where the first variable in the unnested variable is undefined.